### PR TITLE
Configure OAuth settings via the UI

### PR DESCRIPTION
**Pull request by RufusHamade**
This change allows a user to configure the OAuth access  key and secret during the server initialisation process.  It's rather more convoluted than I would've liked so that we can handle the case where the user enters an invalid OAuth key without completely borking the install.

It shouldn't affect the NYT in any way.

### CODE REVIEW

File: packages/frontend-web/src/app/scenes/Settings/components/OAuthConfig.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  return (
+    <div>
+      <p>Configure your connection to the Google Authentication service.</p>
+      <p>
+        To allocate and view tokens, visit <a
+          href="https://console.developers.google.com/apis/credentials"
+          target="_blank"
+          style={{color: 'inherit', fontWeight: 'bold', textDecoration: 'underline'}}
+        >the Google API console</a>.
+        If you haven't yet got an appropriate <b>OAuth Client ID</b> yet, or want to allocate a new one,
```
Comment (sorensenjs) [inline packages/frontend-web/src/app/scenes/Settings/components/OAuthConfig.tsx]: haven't yet got doesn't scan well. and there's two yets. suggest:
If you don't yet have an OAuth Client ID

File: packages/frontend-web/src/app/scenes/Settings/components/OAuthConfig.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        <TextField
+          className={classes.textField}
+          label="Client Secret"
+          value={secret}
+          onChange={secretHandler}
+          margin="normal"
+          variant="outlined"
+        />
+      </div>
+      {props.showLogoutWarning &&
```
Comment (sorensenjs) [inline packages/frontend-web/src/app/scenes/Settings/components/OAuthConfig.tsx]: this && seems weird.

File: packages/frontend-web/src/app/scenes/Settings/components/OAuthConfig.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      <p>
+        The server will take a few seconds to reconfigure itself after the OAuth configuration has changed,
+        and during this time, it will not be available.  If you have login issues, please try again after a few seconds.
+      </p>
+      <div style={{textAlign: 'right', marginTop: '2vh', marginRight: '9vw'}}>
+        <Button
+          variant="contained"
+          color="primary"
+          onClick={submitHandler}
+          disabled={!(id.length > 0 && secret.length > 0)}
```
Comment (sorensenjs) [inline packages/frontend-web/src/app/scenes/Settings/components/OAuthConfig.tsx]: isn't there a more idiomatic way for checking for empty strings?

File: packages/frontend-web/src/app/scenes/Settings/components/OAuthConfig.tsx (hunk 1 above)
Comment (RufusHamade) [reply packages/frontend-web/src/app/scenes/Settings/components/OAuthConfig.tsx]: That is way better.  Thanks for that.

File: packages/frontend-web/src/app/scenes/Settings/components/OAuthConfig.tsx (hunk 2 above)
Comment (RufusHamade) [reply packages/frontend-web/src/app/scenes/Settings/components/OAuthConfig.tsx]: This is kinda the idiomatic way of conditionally including bits of a react tree.  See e.g., 

https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

And I agree with you - I really don't like it either.

**Pull request merged**